codegen/test-utils/src/main/java/software/amazon/smithy/java/codegen/test/PluginTestRunner.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.codegen.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import software.amazon.smithy.build.FileManifest;
+import software.amazon.smithy.build.MockManifest;
+import software.amazon.smithy.build.SmithyBuild;
+import software.amazon.smithy.build.model.SmithyBuildConfig;
+import software.amazon.smithy.model.Model;
+
+public class PluginTestRunner {
+
+    private PluginTestRunner() {}
+
+    private static final Predicate<String> NOT_COMMENT = Predicate.not(Pattern.compile("^ *//.*$").asMatchPredicate());
+
+    public static Optional<String> findGotContent(Path found, TestCase test) {
+        for (var manifest : test.manifests) {
+            var fileInsideBaseDir =
+                    new File(found.toString().replace(manifest.getBaseDir().toString() + "/", "")).toPath();
+            var contents = manifest.getFileString(fileInsideBaseDir);
+            if (contents.isPresent()) {
+                return contents;
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static void assertContentEquals(String left, String right) {
+        try {
+            assertEquals(normalizeSpace(left), normalizeSpace(right));
+        } catch (Throwable e) {
+            assertEquals(left, right);
```

Comment:
IMO we shouldn’t apply any normalization. Codegen must produce identical output, byte-for-byte, every time.

Reply:
I thought I removed this bit as discussed, my bad, let me fix it.